### DISCUSSION TITLE
Restrict invite permissions to user

### DISCUSSION THREAD

**Discussion by etfz** [dani-garcia/vaultwarden#3359] [Ideas]
Hi,

Looking to set up an instance with a single "superuser" that can create organisations and invite users to them. Signups should not be allowed by any other means. I would like for users to be able to create collections within their organisations, but that requires the admin role, which includes invite permissions.

I would therefor like to see a variable controlling who is allowed to send invitations, similar to the `org_creation_users` variable.

I learned that you can invite users via the admin panel, even when invitations are disabled globally, but that is not ideal.

**Comment by stefan0xC** [comment]
> that requires the admin role

Well, there's also a `Manager` role to manage collections. You might want to check out [this Bitwarden help page about the available user types](https://bitwarden.com/help/user-types-access-control/#member-roles). (Note: At the moment vaultwarden [does not support custom roles](https://github.com/dani-garcia/vaultwarden/wiki#missing-features).)

**Reply by etfz** [reply]
Ah, I did not realise managers were able to create collections. That is not what the web vault description seemed to imply.
> Managers can access and manage assigned collections in your organization.

Still, I am interested in essentially all the admin permissions except for invitations.

**Reply by stefan0xC** [reply]
> Still, I am interested in essentially all the admin permissions except for invitations.

Well, as you already know you can disable invitations.

**Comment by etfz** [comment]
Would a PR be accepted?

**Reply by BlackDex** [reply]
I would not like to add more features like this. It only makes it more complex.
If you really do not want people to signup or allow invites, then just disable it all, and invite people via via the admin interface.
In the end you still need someone to invite someone into an org.

Also, if anything into regards of limited role access, i would like to have the custom roles working.
That allows a more fine grained access, including a checkbox to allow editing members, which if unchecked doesn't allow that person to see the member list, and also do not invite them. They can however add them to groups or collections if allowed.

**Reply by etfz** [reply]
Fair enough.

The problem with inviting through the admin panel is you can't actually invite someone _to an organization_, as far as I can tell, and I don't necessarily want to share the admin password. If you could invite to an organization via the admin panel that could be a workable compromise. I don't know whether there's anything preventing that.

Custom roles would be cool. It might be slightly more inconvenient because I would need to set up the same custom permissions for each organization admin, but I suppose it's the best addition to the project.
